### PR TITLE
fix(useWatch): default value for array of inputs

### LINKED ISSUE #9530: issue: error when using `useWatch` with a name array and default values as object
### Version Number

v7.40.0

### Codesandbox/Expo snack

https://stackblitz.com/edit/react-ts-beok4h?file=App.tsx

### Steps to reproduce

1. Modify the TS doc example for `useWatch` so it uses an array as `name`, as mentioned in the comment next to it
2. Update `defaultValues` to an object following TS directives

```ts
function FirstNameWatched({ control }: { control: Control<FormInputs> }) {
  const [firstName, lastName] = useWatch({
    control,
    name: ['firstName', 'lastName'],
    defaultValue: { firstName: 'default', lastName: 'default' },
  });

  return (
    <p>
      Watch: {firstName} {lastName}
    </p>
  );
}
```
3. See the runtime error
```
object is not iterable (cannot read property Symbol(Symbol.iterator))
```

No errors when passing an array as defaultValues but then it generates other TS errors. 

### Expected behaviour

Should not crash, or type should be updated 

### What browsers are you seeing the problem on?

_No response_

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by adesurirey**
Closes #9530

### CODE REVIEW

File: src/useWatch.ts [hunk 1]
```diff
@@ -183,7 +183,10 @@ export function useWatch<TFieldValues extends FieldValues>(
   const [value, updateValue] = React.useState<unknown>(
     isUndefined(defaultValue)
       ? control._getWatch(name as InternalFieldName)
-      : defaultValue,
+      : control._getWatch(
+          name as InternalFieldName,
+          defaultValue as DeepPartialSkipArrayKey<TFieldValues>,
+        ),
   );
```
Comment (bluebill1049) [inline src/useWatch.ts:190]: I think we can simplify to the following

```tsx
const [value, updateValue] = React.useState<unknown>(
  control._getWatch(
    name as InternalFieldName,
    defaultValue as DeepPartialSkipArrayKey<TFieldValues>,
  ),
);
```

**Review by bluebill1049 — COMMENTED** [review]
Looking great! thanks a lot for the fix.

**Review by bluebill1049 — APPROVED** [review]
Let's go! Thanks.

**Comment by adesurirey** [comment]
Thank you @bluebill1049!

**Comment by adesurirey** [comment]
@bluebill1049 when will it be released?

**Comment by bluebill1049** [comment]
I am planning to do the final release of the year next week.

**Pull request merged**